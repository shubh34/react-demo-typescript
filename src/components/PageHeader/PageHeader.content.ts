export type contentType = {
    pageHeaderTitle: string,
    homeHeaderTitle:  string,
    addBlogHeaderTitle:  string,
}
const content : contentType = {
    pageHeaderTitle: 'My Blogs',
    homeHeaderTitle: 'Home',
    addBlogHeaderTitle: 'Add Blog'

}

export default content;