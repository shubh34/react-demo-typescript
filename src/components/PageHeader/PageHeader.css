.nav {
    background-color: white;
    position: fixed;
    top: 0;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    max-width: 1280px;
    z-index: 100;
   
  }
  .nav .nav_section_header {
    padding: 16px;
    margin : 0px
  }
  .nav_search_box {
      display: none;
     
  }
  .nav_link {
    border: 2px solid;
    padding: 8px;
    border-radius: 8px;
  }
  .nav_link:hover, .nav_link--active {
    background-color: #4183c4;
    color:white;
  }
  @media only screen and (min-width: 768px) {
    /* Remove this when searchnis implemented  */
    /* .nav_search_box {
        display: inline-block;
    } */
  }
  ul {
    list-style-type: none;
    padding-right:16px 
  }
  li {
    display: inline-block;
    padding: 0 8px;
  }