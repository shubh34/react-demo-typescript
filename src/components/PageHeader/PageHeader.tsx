import './PageHeader.css';
import React from 'react';
import { NavLink } from 'react-router-dom';
import { Icon, Search } from 'semantic-ui-react';
import SectionHeader from '../sharedComponents/SectionHeader/SectionHeader';

import pageHeaderContent from './PageHeader.content';

const PageHeader: React.FC = () => {
	const { homeHeaderTitle, addBlogHeaderTitle, pageHeaderTitle } = pageHeaderContent;
	return (
		<nav className='nav'>
			<SectionHeader header={pageHeaderTitle} className='nav_section_header' />
			<ul>
				<li>
					<NavLink className='nav_link' exact activeClassName='nav_link--active' to='/'>
						{homeHeaderTitle}
					</NavLink>
				</li>
				<li>
					<NavLink className='nav_link' activeClassName='nav_link--active' to='/addBlog'>
						<Icon name='plus' />
						{addBlogHeaderTitle}
					</NavLink>
				</li>
				<li className='nav_search_box'>
					<Search loading={false} />
				</li>
			</ul>
		</nav>
	);
};

export default PageHeader;
