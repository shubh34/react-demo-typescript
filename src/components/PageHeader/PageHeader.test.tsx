import React from 'react';
import { shallow } from 'enzyme';
import PageHeader from './PageHeader';

it('renders without crashing', () => {
	const wrapper = shallow(<PageHeader />);
	expect(wrapper).toMatchSnapshot();
});
