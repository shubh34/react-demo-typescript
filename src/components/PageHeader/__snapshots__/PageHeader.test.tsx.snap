// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<nav
  className="nav"
>
  <SectionHeader
    className="nav_section_header"
    header="My Blogs"
  />
  <ul>
    <li>
      <NavLink
        activeClassName="nav_link--active"
        className="nav_link"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="nav_link--active"
        className="nav_link"
        to="/addBlog"
      >
        <Icon
          as="i"
          name="plus"
        />
        Add Blog
      </NavLink>
    </li>
    <li
      className="nav_search_box"
    >
      <Search
        icon="search"
        input="text"
        loading={false}
        minCharacters={1}
        noResultsMessage="No results found."
        showNoResults={true}
      />
    </li>
  </ul>
</nav>
`;
